src/Albany_RegressionTests.cpp

```diff
@@ -41,77 +41,71 @@ checkSolveTestResults(
-                                     << ") greater than number of responses ("
-                                     << g->space()->dim() << ") !");
-    Teuchos::Array<double> testValues =
-        testParams->get<Teuchos::Array<double>>("Test Values");
-
-    ALBANY_ASSERT(
-        numResponseTests == testValues.size(),
-        "Number of Response Tests (" << numResponseTests
-                                     << ") != number of Test Values ("
-                                     << testValues.size() << ") !");
-
-    Teuchos::ArrayRCP<const ST> g_view = getLocalData(g);
-    for (int i = 0; i < testValues.size(); i++) {
-      auto s = std::string("Response Test ") + std::to_string(i);
-      failures += scaledCompare(g_view[i], testValues[i], relTol, absTol, s);
-      comparisons++;
-    }
+  // Test the response (g)
+  const int numResponseTests = 1;
+  ALBANY_ASSERT(
+      g != Teuchos::null,
+      "There is Response Test but the response vector is null!");
+  ALBANY_ASSERT(
+      numResponseTests <= g->space()->dim(),
+      "Number of Response Test (" << numResponseTests
+                                    << ") greater than number of responses ("
+                                    << g->space()->dim() << ") !");
+  double testValue =
+      testParams->get<double>("Test Value");
+
+  Teuchos::ArrayRCP<const ST> g_view = getLocalData(g);
+  auto s = std::string("Response Test ") + std::to_string(response_index);
+  failures += scaledCompare(g_view[0], testValue, relTol, absTol, s);
+  comparisons++;
+
+  if (parameter_index == -1) {
+    storeTestResults(testParams, failures, comparisons);
+
+    return failures;
   }
 
   // Repeat comparisons for sensitivities
   Teuchos::ParameterList* sensitivityParams = 0;
   std::string             sensitivity_sublist_name =
-      strint("Sensitivity Comparisons", parameter_index);
-  if (parameter_index == 0 && !testParams->isSublist(sensitivity_sublist_name))
-    sensitivityParams = testParams;
-  else if (testParams->isSublist(sensitivity_sublist_name))
+      strint("Sensitivity For Parameter", parameter_index);
+  if (testParams->isSublist(sensitivity_sublist_name))
     sensitivityParams = &(testParams->sublist(sensitivity_sublist_name));
-  int numSensTests = 0;
-  if (sensitivityParams != 0) {
-    numSensTests =
-        sensitivityParams->get<int>("Number of Sensitivity Comparisons", 0);
-  }
-  if (numSensTests > 0) {
-    ALBANY_ASSERT(
-        dgdp != Teuchos::null,
-        "There are Sensitivity Tests but the sensitivity vector ("
-            << response_index << ", " << parameter_index << ") is null!");
-    if (numSensTests != dgdp->range()->dim()) {
-      ALBANY_ASSERT(
-          false,
-          "Number of sensitivity tests ("
-              << numSensTests << ") != number of sensitivities ["
-              << response_index << "][" << parameter_index << "] ("
-              << dgdp->range()->dim() << ") !");
-    }
+  else {
```

Comment:
Not sure what `storeTestResults` does, but I think we should not error out if we do not have the sensitivity associated to a parameter. It's fine, we don't  want to require that we check all sensitivities.

Reply:
This does not error out if we do not have the sensitivity associated to a parameter.
This, instead, does not add sensitivity checks if the parameter list does not include the sublist associated with the current parameter.

I think that the `return failures;` is misleading in this case.
The `failures` variable is an integer which corresponds to the number of failed comparisons. This number is zero when all the tests passed.